fbw-a32nx/src/systems/fmgc/src/efis/EfisSymbols.ts

```diff
@@ -187,196 +198,237 @@ export class EfisSymbols<T extends number> {
     const vnavPredictionsChanged = this.lastVnavDriverVersion !== this.guidanceController.vnavDriver.version;
     this.lastVnavDriverVersion = this.guidanceController.vnavDriver.version;
 
-    const hasSuitableRunway = (airport: Airport): boolean =>
-      airport.longestRunwayLength >= 1500 && airport.longestRunwaySurfaceType === RunwaySurfaceType.Hard;
-
-    for (const side of EfisSymbols.sides) {
-      const range = this.rangeValues[SimVar.GetSimVarValue(`L:A32NX_EFIS_${side}_ND_RANGE`, 'number')];
-      const mode: EfisNdMode = SimVar.GetSimVarValue(`L:A32NX_EFIS_${side}_ND_MODE`, 'number');
-      const efisOption = SimVar.GetSimVarValue(`L:A32NX_EFIS_${side}_OPTION`, 'Enum');
-
-      const rangeChange = this.lastRange[side] !== range;
-      this.lastRange[side] = range;
-      const modeChange = this.lastMode[side] !== mode;
-      this.lastMode[side] = mode;
-      const efisOptionChange = this.lastEfisOption[side] !== efisOption;
-      this.lastEfisOption[side] = efisOption;
-      const nearbyOverlayChanged = (efisOption & ~EfisOption.Constraints) > 0 && nearbyFacilitiesChanged;
-      const efisInterfaceChanged = this.lastEfisInterfaceVersions[side] !== this.efisInterfaces[side].version;
-      this.lastEfisInterfaceVersions[side] = this.efisInterfaces[side].version;
+    // const hasSuitableRunway = (airport: Airport): boolean =>
+    //   airport.longestRunwayLength >= 1500 && airport.longestRunwaySurfaceType === RunwaySurfaceType.Hard;
+
+    const range = this.rangeValues[SimVar.GetSimVarValue(`L:A32NX_EFIS_${this.side}_ND_RANGE`, 'number')];
+    const efisOption = SimVar.GetSimVarValue(`L:A32NX_EFIS_${this.side}_OPTION`, 'Enum');
+
+    const rangeChange = this.lastRange !== range;
+    this.lastRange = range;
+    const modeChange = this.lastMode !== mode;
+    this.lastMode = mode;
+    const efisOptionChange = this.lastEfisOption !== efisOption;
+    this.lastEfisOption = efisOption;
+    this.nearbyFacilitesChanged &&= (efisOption & ~EfisOption.Constraints) > 0;
+    const efisInterfaceChanged = this.lastEfisInterfaceVersion !== this.efisInterface.version;
+    this.lastEfisInterfaceVersion = this.efisInterface.version;
+
+    const editArea = this.calculateEditArea(range, mode);
+    // MSFS2024 can load facs much more efficiently with minimal facilities, so we can afford to follow the MRP in plan mode.
+    const nearbyCentre = isMsfs2024() ? mapReferencePoint : ppos;
+    const nearbyRadius = Math.max(...editArea);
+    if (efisOption & EfisOption.Airports) {
```

Comment:
Actually can be a little smarter here and find a point in the centre of the edit area, and reduce the radius to just cover the corners of the edit area.

Reply:
Done.